src/Stratis.Bitcoin.Features.PoA/Payloads/PoAHeadersPayload.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using NBitcoin;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using TracerAttributes;
+
+namespace Stratis.Bitcoin.Features.PoA.Payloads
+{
+    /// <summary>
+    /// Block headers received after a getheaders messages.
+    /// </summary>
+    [Payload("poahdr")]
+    public class PoAHeadersPayload : Payload
+    {
+        private List<PoABlockHeader> headers;
+
+        public List<PoABlockHeader> Headers
+        {
+            get { return this.headers; }
+        }
+
+        public PoAHeadersPayload()
+        {
+            this.headers = new List<PoABlockHeader>();
+        }
+
+        public PoAHeadersPayload(params PoABlockHeader[] headers)
+        {
+            this.Headers.AddRange(headers);
+        }
+
+        [NoTrace]
+        public override void ReadWriteCore(BitcoinStream stream)
+        {
+            if (stream.Serializing)
```

Comment:
why do we need to do all this?
The readWrite method does it all by itself I think.
https://github.com/stratisproject/StratisBitcoinFullNode/blob/d3a1952a85a47d3a041c137f970162dd5a3e5da9/src/NBitcoin/BitcoinStream.cs#L215-L248

Looks like we can just have:
```
[NoTrace]
public override void ReadWriteCore(BitcoinStream stream)
{
    stream.ReadWrite(ref this.headers);
}
```

Reply:
you are right. fixed